pkg/deployments/tasks/20220413-polygon-root-gauge-factory/test/task.fork.ts

```diff
@@ -0,0 +1,182 @@
+    await authorizer
+      .connect(govMultisig)
+      .grantRole(await authorizerAdaptor.getActionId(gauge.interface.getSighash('checkpoint')), admin.address);
+  });
+
+  it('add gauge to gauge controller', async () => {
+    await gaugeAdder.addGaugeFactory(factory.address, 3); // Polygon is Gauge Type 3
+    await gaugeAdder.addPolygonGauge(gauge.address);
+
+    expect(await gaugeController.gauge_exists(gauge.address)).to.be.true;
+  });
+
+  it('vote for gauge', async () => {
+    expect(await gaugeController.get_gauge_weight(gauge.address)).to.equal(0);
+    await gaugeController.connect(veBALHolder).vote_for_gauge_weights(gauge.address, 10000); // Max voting power is 10k points
+
+    // We now need to go through an epoch for the votes to be locked in
+    await advanceTime(DAY * 8);
+
+    await gaugeController.checkpoint();
+    expect(await gaugeController['gauge_relative_weight(address)'](gauge.address)).to.be.gt(0);
+  });
+
+  it('mint & bridge tokens', async () => {
+    // The gauge has votes for this week, and it will mint the first batch of tokens. We store the current gauge
+    // relative weight, as it will change as time goes by due to vote decay.
+    const firstMintWeekTimestamp = await currentWeekTimestamp();
+    const gaugeRelativeWeight = await gaugeController['gauge_relative_weight(address)'](gauge.address);
+
+    const calldata = gauge.interface.encodeFunctionData('checkpoint');
+
+    // Even though the gauge has relative weight, it cannot mint yet as it needs for the epoch to finish
+    const zeroMintTx = await authorizerAdaptor.connect(admin).performAction(gauge.address, calldata);
+    expectEvent.inIndirectReceipt(await zeroMintTx.wait(), gauge.interface, 'Checkpoint', {
+      periodTime: firstMintWeekTimestamp.sub(WEEK), // Process past week, which had zero votes
+      periodEmissions: 0,
+    });
+    // No token transfers are performed if the emissions are zero, but we can't test for a lack of those
+
+    await advanceTime(WEEK);
+
+    // The gauge should now mint and send all minted tokens to the Polygon bridge
+    const mintTx = await authorizerAdaptor.connect(admin).performAction(gauge.address, calldata);
+    const event = expectEvent.inIndirectReceipt(await mintTx.wait(), gauge.interface, 'Checkpoint', {
+      periodTime: firstMintWeekTimestamp,
+    });
+    const actualEmissions = event.args.periodEmissions;
+
+    // The amount of tokens minted should equal the weekly emissions rate times the relative weight of the gauge
+    const weeklyRate = (await BALTokenAdmin.getInflationRate()).mul(WEEK);
+
+    const expectedEmissions = gaugeRelativeWeight.mul(weeklyRate).div(FP_SCALING_FACTOR);
+    expectEqualWithError(actualEmissions, expectedEmissions, 0.001);
+
+    // Tokens are minted for the gauge
+    expectEvent.inIndirectReceipt(await mintTx.wait(), BAL.interface, 'Transfer', {
+      from: ZERO_ADDRESS,
+      to: gauge.address,
+      value: actualEmissions,
+    });
```

Comment:
We have so many of these littered about we should really wrap it up into an `expectTransferEvent` globally at some point.

Reply:
We do use `expectTokenTransfer` extensively, but our forking infrastructure and associated helpers could really use some help, yes (as can be seen in the huge task loading boilerplate).